torch/_functorch/aot_autograd.py

```diff
@@ -824,6 +829,11 @@ def inner(*flat_args):
             if not isinstance(o, torch.Tensor):
                 output_type = OutputType.non_alias
                 base_idx = None
+            # Need to check for both custom cpp (CppFunction) and python (BackwardCFunction) autograd fns
+            elif curr_storage in inp_storage_refs and o.grad_fn is not None \
+                    and type(o.grad_fn).__name__ in ["CppFunction", "BackwardCFunction"]:
+                output_type = OutputType.custom_function_view
+                base_idx = None
```

Comment:
cc @zou3519 finally updated and added some C++ tests (I ended up co-opting the open registration test file, since I found it very convenient for authoring custom ops in C++ that I could run torch.compile on).

Good catch mentioning that we should check for C++ autograd functions that alias too - they needed explicit handling (because cpp and python grad_fn's have different python types).

Reply:
Nice!